modular_skyrat/modules/primitive_fun/code/cutting_board.dm

```diff
@@ -0,0 +1,125 @@
+
+/obj/item/cutting_board/Initialize(mapload)
+	. = ..()
+	if(processor_inputs)
+		return
+	processor_inputs = list()
+	for(var/datum/food_processor_process/recipe as anything in subtypesof(/datum/food_processor_process)) //this is how tg food processors do it just in case this is digusting
+		if(!initial(recipe.input))
+			continue
+		recipe = new recipe
+		var/list/typecache = list()
+		var/list/bad_types
+		for(var/bad_type in recipe.blacklist)
+			LAZYADD(bad_types, typesof(bad_type))
+		for(var/input_type in typesof(recipe.input) - bad_types)
+			typecache[input_type] = recipe
+		for(var/machine_type in typesof(recipe.required_machine))
+			LAZYADD(processor_inputs[machine_type], typecache)
+
+/obj/item/cutting_board/update_appearance()
+	. = ..()
+	cut_overlays()
+	if(!length(contents))
+		return
+	var/image/overlayed_item = image(icon = contents[1].icon, icon_state = contents[1].icon_state, pixel_y = 2)
+	add_overlay(overlayed_item)
+
+/obj/item/cutting_board/examine(mob/user)
+	. = ..()
+	. += span_notice("You can process food similar to a food processor by putting food on this and using a <b>knife</b> on it.")
+	. += span_notice("It can be (un)secured with <b>Right Click</b>")
+	. += span_notice("You can make it drop its item with <b>Alt Click</b>")
+	if(length(contents))
+		. += span_notice("It has [contents[1]] sitting on it.")
+
+/obj/item/cutting_board/Destroy()
+	drop_everything_contained()
+	return ..()
+
+/obj/item/cutting_board/AltClick(mob/user)
+	if(!length(contents))
+		balloon_alert(user, "nothing on board")
+		return
+	drop_everything_contained()
+	balloon_alert(user, "cleared board")
+
+///Drops all contents at the turf of the item
+/obj/item/cutting_board/proc/drop_everything_contained()
+	if(!length(contents))
+		return
+	for(var/obj/target_item as anything in contents)
+		target_item.forceMove(get_turf(src))
+
+/obj/item/cutting_board/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+	if(!can_interact(user) || !user.canUseTopic(src, be_close = TRUE))
+		return
+	set_anchored(!anchored)
+	balloon_alert_to_viewers(anchored ? "secured" : "unsecured")
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+///Takes the given obj (processed thing) and gets its results from the recipe list, spawning the results and deleting the original obj
+/obj/item/cutting_board/proc/process_food(datum/food_processor_process/recipe, obj/processed_thing)
+	if(!recipe.output || !loc || QDELETED(src))
+		return
+	var/food_multiplier = recipe.food_multiplier
+	for(var/i in 1 to food_multiplier)
+		var/obj/new_food_item = new recipe.output(drop_location())
+		new_food_item.pixel_x = rand(-6, 6)
+		new_food_item.pixel_y = rand(-6, 6)
+		if(!processed_thing.reagents) //backup in case we really fuck up
+			continue
+		processed_thing.reagents.copy_to(new_food_item, processed_thing.reagents.total_volume, multiplier = 1 / food_multiplier)
+	qdel(processed_thing)
+	update_appearance()
```

Comment:
```suggestion
	if(!recipe.output || !loc || QDELETED(src))
		return
	
	var/food_multiplier = recipe.food_multiplier
	for(var/i in 1 to food_multiplier)
		var/obj/new_food_item = new recipe.output(drop_location())
		new_food_item.pixel_x = rand(-6, 6)
		new_food_item.pixel_y = rand(-6, 6)
		
		if(!processed_thing.reagents) //backup in case we really fuck up
			continue
		
		processed_thing.reagents.copy_to(new_food_item, processed_thing.reagents.total_volume, multiplier = 1 / food_multiplier)
	
	qdel(processed_thing)
	update_appearance()
```